src/extensions/wcf/Microsoft.DotNet.UpgradeAssistant.Extensions.WCFUpdater/WCFUpdateStep.cs

```diff
@@ -0,0 +1,226 @@
+
+        public override IEnumerable<string> DependsOn { get; } = new[]
+        {
+            // Project should be backed up before changing source
+            WellKnownStepIds.BackupStepId,
+
+            // Template files should be added prior to changing source (since some code fixers will change added templates)
+            WellKnownStepIds.TemplateInserterStepId,
+
+            // Project should have correct TFM
+            WellKnownStepIds.SetTFMStepId,
+
+            // Project's config file, source code, and project file need to be updated first before adding CoreWCF services
+            WellKnownStepIds.ConfigUpdaterStepId,
+
+            WellKnownStepIds.PackageUpdaterStepId
+        };
+
+        public override IEnumerable<string> DependencyOf { get; } = new[]
+        {
+            WellKnownStepIds.NextProjectStepId,
+        };
+
+        public WCFUpdateStep(ILogger<WCFUpdateStep> logger)
+            : base(logger)
+        {
+            if (logger is null)
+            {
+                throw new ArgumentNullException(nameof(logger));
+            }
+        }
+
+        protected override Task<bool> IsApplicableImplAsync(IUpgradeContext context, CancellationToken token) => Task.FromResult(context?.CurrentProject is not null);
+
+        protected override Task<UpgradeStepInitializeResult> InitializeImplAsync(IUpgradeContext context, CancellationToken token)
+        {
+            if (context is null)
+            {
+                throw new ArgumentNullException(nameof(context));
+            }
+
+            try
+            {
+                var currentProject = context.CurrentProject.Required();
+                return Initialize(currentProject);
+            }
+            catch (Exception e)
+            {
+                Logger.LogError(e, "Unexpected error happened during CoreWCF porting.");
+                return Task.FromResult(new UpgradeStepInitializeResult(UpgradeStepStatus.Failed, "Project failed to port to CoreWCF. Please review error message and log file for more information.", BuildBreakRisk.Medium));
+            }
+        }
+
+        protected override Task<UpgradeStepApplyResult> ApplyImplAsync(IUpgradeContext context, CancellationToken token)
+        {
+            if (context is null)
+            {
+                throw new ArgumentNullException(nameof(context));
+            }
+
+            try
+            {
+                return Apply();
+            }
+            catch (Exception e)
+            {
+                Logger.LogError(e, "Unexpected error happened during CoreWCF porting.");
+                return Task.FromResult(new UpgradeStepApplyResult(UpgradeStepStatus.Failed, "Project failed to port to CoreWCF. Please review error message and log file for more information."));
+            }
+        }
+
+        public Task<UpgradeStepInitializeResult> Initialize(IProject project)
+        {
+            _path = new Dictionary<string, IEnumerable<string>>();
```

Comment:
A class with fields for expected paths might be better than a dictionary. More strongly typed.